src/codegen/riscv64/macro-assembler-riscv64.cc

```diff
@@ -3645,17 +3645,20 @@ void TurboAssembler::MulOverflow(Register dst, Register left,
          overflow != scratch);
   DCHECK(overflow != left && overflow != right_reg);
 
+
+  addiw(left, left, 0);
+  addiw(right_reg, right_reg, 0);
```

Comment:
Is it okay to overwrite these registers in here?

Reply:
My purpose is to ensure that the register is a 32-bit signed integer.Because in test case test-macro-assembler-riscv64/OverflowInstructions, the input register is int64_t.
For better representation, I use sext_w instead.

Reply:
Right, I understand that the numbers should be sign-extended at 32-bits, but I'm not sure if its okay to modify those inputs here.

Reply:
> Right, I understand that the numbers should be sign-extended at 32-bits, but I'm not sure if its okay to modify those inputs here.
Because muloverflow is a 32-bit integer multiplication, but i use mul is 64-bit mul. In order to ensure that the normal calculation result is 32bit, I extend the input parameters to 32 bit symbols. i alter code for this:
```
  sext_w(left, left);
  sext_w(right_reg, right_reg);

  mul(overflow, left, right_reg);
  sext_w(dst, overflow);
  xor_(overflow, overflow, dst);
```

if dst is not equal to overflow, it Indicates that an overflow has occurred.

Reply:
Sorry, I got that wrong. I changed the code and saved the input using push / pop.

Reply:
I add a scratch for MulOverflow.